### PR TITLE
fix: Fixed the problem of incorrect update of select ellipsisTrigger …

### LINKED ISSUE #1862: [BUG] Select ellipsisTrigger incorrectly updates under certain boundary conditions
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Which Component

Select

### Semi Version

latest

### Current Behavior

1. 使用 250px 的宽度，选中、取消选中 「抖音」后，tag 更新错误，应为「西瓜视频」「轻颜相机」「+2」

2. 使用 220px 的宽度，取消选中 「西瓜视频」后，tag 更新错误，应为 「轻颜相机」「剪映」

### Steps To Reproduce

20231018174035_rec_

20231018174144_rec_

### ReproducibleCode

```markdown
import React from 'react';
import { Select } from '@douyinfe/semi-ui';

() => (
    <>
        <Select
            multiple
            maxTagCount={3} 
            showRestTagsPopover={true}
            restTagsPopoverProps={{ position: 'top' }}
            style={{ width: '250px' }}
            // style={{ width: '220px' }}
            defaultValue={['xigua', 'ulikecam', 'jianying']}
            ellipsisTrigger
        >
            <Select.Option value="abc">抖音</Select.Option>
            <Select.Option value="ulikecam">轻颜相机</Select.Option>
            <Select.Option value="jianying">剪映</Select.Option>
            <Select.Option value="xigua">西瓜视频</Select.Option>
        </Select>
    </>
);
```

### Environment

```markdown
- OS:
- browser:
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by YannLynn**
…under certain boundary conditions

[中文模板 / Chinese Template](https://github.com/DouyinFE/semi-design/blob/main/.github/PULL_REQUEST_TEMPLATE.zh-CN.md)

- [x] I have read and followed [Pull Request Guidelines](https://github.com/DouyinFE/semi-design/blob/main/CONTRIBUTING-en-US.md#pull-request-guidelines) of the contributing guide.

### What kind of change does this PR introduce? (check at least one)

 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update
 - [ ] Refactor
 - [ ] Test Case
 - [ ] TypeScript definition update
 - [ ] Document improve
 - [ ] CI/CD improve
 - [ ] Branch sync
 - [ ] Other, please describe:

### PR description

Fixes #1862
在两种边界条件下会导致计算错误
1. 最后一项为被截断的 tag。
2. 无截断 tag，宽度不足，选择项数目从等于 maxTagCount 到 maxTagCount + 1。

**出错原因**
前者涉及到 Overflow 的内部计算，其计算溢出项是会判断当前项目长度和容器长度的关系，选择前被截断的 tag 会被当作溢出项首先被加入宽度中，但是也有可能被当作正常项目**再次**被加入宽度计算，导致被计算两遍。

后者，因为 Overflow 输入的 items 始终为前 maxTagCount 项值，故选择后 items 参数值不变，不会触发溢出更新。

### Changelog
🇨🇳 Chinese
- Fix: 修复 select ellipsisTrigger 在某些边界条件下更新不正确的问题

---

🇺🇸 English
- Fix: Fixed the problem of incorrect update of select ellipsisTrigger under certain boundary conditions

### Checklist
- [x] Test or no need
- [x] Document or no need
- [x] Changelog or no need

### Other
- [ ] Skip Changelog

### CODE REVIEW

File: cypress/e2e/select.spec.js [hunk 1]
```diff
@@ -62,6 +62,34 @@ describe('Select', () => {
         cy.wait(500);
         cy.get('.semi-select-option').should('have.text', 'Design');
     });
+
+    it.only('ellipsisTrigger', () => {
```
Comment (YyumeiZhang) [inline cypress/e2e/select.spec.js:66]: 这个 only 需要删除下

**Review by YyumeiZhang — APPROVED** [review]
